### PR TITLE
DMP-3960 Swap batch logic - automated_task vs application.yaml

### PR DESCRIPTION
**Pull request by karen-hedges**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/DMP-3960

### Change description ###

Swapped the batching around so that the number of EOD's retrieved is based on the batch size in the database and the number of items to write in the manifest is yaml config driven.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/darts/arm/config/UnstructuredToArmProcessorConfiguration.java [hunk 1]
```diff
@@ -13,6 +13,7 @@
 @Configuration
 public class UnstructuredToArmProcessorConfiguration {
 
-    private int maxResultSize;
+    private int maxArmManifestItems;
```
Comment (cmagowan-hmcts) [inline src/main/java/uk/gov/hmcts/darts/arm/config/UnstructuredToArmProcessorConfiguration.java:16]: these dont exist, they are in a different place in the yaml. probably need to change line 10.
Also, max-result-size is still int eh yaml, and so should be removed.

File: src/main/java/uk/gov/hmcts/darts/arm/service/impl/UnstructuredToArmBatchProcessorImpl.java [hunk 2]
```diff
@@ -67,23 +66,29 @@ public void processUnstructuredToArm(int armBatchSize) {
...
-        if (!eodsForTransfer.isEmpty()) {
-            //ARM has a max batch size, so lets loop through the big list creating lots of individual batches for ARM to process separately
-            List<List<ExternalObjectDirectoryEntity>> batchesForArm = ListUtils.partition(eodsForTransfer, armBatchSize);
-            int batchCounter = 1;
-            UserAccountEntity userAccount = userIdentity.getUserAccount();
-            for (List<ExternalObjectDirectoryEntity> eodsForBatch : batchesForArm) {
-                log.info("Creating batch {} out of {}", batchCounter++, batchesForArm.size());
-                createAndSendBatchFile(eodsForBatch, userAccount);
+        int querySize = unstructuredToArmProcessorConfiguration.getMaxArmManifestItems();
+        if (armBatchSize < unstructuredToArmProcessorConfiguration.getMaxArmManifestItems()) {
```
Comment (cmagowan-hmcts) [inline src/main/java/uk/gov/hmcts/darts/arm/service/impl/UnstructuredToArmBatchProcessorImpl.java:70]: `armBatchSize` should be renamed, as its no longer the arm batch size, its the total number of items we intent to process.

File: src/main/java/uk/gov/hmcts/darts/arm/config/UnstructuredToArmProcessorConfiguration.java (hunk 1 above)
Comment (cmagowan-hmcts) [inline src/main/java/uk/gov/hmcts/darts/arm/config/UnstructuredToArmProcessorConfiguration.java:16]: is `maxArmManifestItems` not referred to the arm batch size?

File: src/main/java/uk/gov/hmcts/darts/arm/service/impl/UnstructuredToArmBatchProcessorImpl.java [hunk 3]
```diff
@@ -67,23 +66,29 @@ public void processUnstructuredToArm(int armBatchSize) {
...
+            querySize = armBatchSize;
+        }
+
+        UserAccountEntity userAccount = userIdentity.getUserAccount();
+
+        for (int batchCounter = 1; batchCounter <= armBatchSize; batchCounter += querySize) {
+            log.info("Batching rows from {} out of a potential rows {}", batchCounter, armBatchSize);
+            List<ExternalObjectDirectoryEntity> batchedEods =
+                unstructuredToArmHelper.getEodEntitiesToSendToArm(eodSourceLocation, EodHelper.armLocation(), querySize);
```
Comment (cmagowan-hmcts) [inline src/main/java/uk/gov/hmcts/darts/arm/service/impl/UnstructuredToArmBatchProcessorImpl.java:79]: the whole point of the previous code was to run this long-running query once, which you have reverted. please reinstate.

File: src/main/resources/db/migration/common/V1_381__update_batch_size_for_automated_task.sql [hunk 4]
```diff
@@ -0,0 +1,2 @@
+update automated_task set batch_size = 100000
```
Comment (hemantasharma1129) [inline src/main/resources/db/migration/common/V1_381__update_batch_size_for_automated_task.sql:1]: @cmagowan-hmcts this need to be in synch with migration team.

**Pull request merged**